packages/lib/modules/pool/actions/create/modal/useCreatePoolInput.ts

```diff
@@ -18,39 +19,93 @@ export function useCreatePoolInput(chainId: number) {
-  const isStablePool = validatePoolType.isStablePool(poolType)
+  if (!poolTokens[0]?.address || !poolTokens[1]?.address) {
+    throw new Error('Pool token address missing for pool creation')
+  }
+
+  const areTokensInOrder =
+    poolTokens[0]?.address?.toLowerCase() < poolTokens[1]?.address?.toLowerCase()
 
   const chain = getGqlChain(chainId)
   const { tokens } = getNetworkConfig(chain)
   const nativeAsset = tokens.nativeAsset.address
   const wNativeAsset = tokens.addresses.wNativeAsset
 
-  const createPoolInput = {
+  const baseInput: CreatePoolV3BaseInput = {
     chainId,
     protocolVersion: 3 as const,
-    poolType,
     name,
     symbol,
     swapFeePercentage: parseUnits(swapFeePercentage, PERCENTAGE_DECIMALS),
-    swapFeeManager,
-    pauseManager,
+    swapFeeManager: swapFeeManager ? swapFeeManager : zeroAddress,
+    pauseManager: pauseManager ? pauseManager : zeroAddress,
     enableDonation,
-    poolHooksContract,
+    poolHooksContract: poolHooksContract ? poolHooksContract : zeroAddress,
     disableUnbalancedLiquidity,
-    ...(isStablePool && { amplificationParameter: BigInt(amplificationParameter) }),
-    tokens: poolTokens.map(({ address, rateProvider, weight, paysYieldFees }) => {
+    tokens: poolTokens.map(({ address, rateProvider, paysYieldFees }) => {
+      const tokenAddress = address === nativeAsset ? wNativeAsset : address
+      if (!tokenAddress) throw new Error('token address missing for pool creation')
       return {
-        address: address === nativeAsset ? wNativeAsset : address,
+        address: tokenAddress,
         tokenType: rateProvider === zeroAddress ? TokenType.STANDARD : TokenType.TOKEN_WITH_RATE,
-        rateProvider,
+        rateProvider: rateProvider ? rateProvider : zeroAddress,
         paysYieldFees,
-        ...(isWeightedPool && weight ? { weight: parseUnits(weight, PERCENTAGE_DECIMALS) } : {}),
       }
     }),
-  } as CreatePoolInput // TODO: solve type complaint for inputs allowing empty strings ""
+  }
+
+  if (poolType === PoolType.Stable || poolType === PoolType.StableSurge) {
+    return {
+      ...baseInput,
+      poolType,
+      amplificationParameter: BigInt(amplificationParameter),
+    }
+  }
+
+  if (poolType === PoolType.Weighted) {
+    return {
+      ...baseInput,
+      poolType,
+      tokens: baseInput.tokens.map((token, index) => ({
+        ...token,
+        weight: parseUnits(poolTokens[index].weight!, PERCENTAGE_DECIMALS),
+      })),
+    }
+  }
+
+  if (poolType === PoolType.ReClamm) {
+    const {
+      initialMinPrice,
+      initialMaxPrice,
+      initialTargetPrice,
+      priceShiftDailyRate,
+      centerednessMargin,
+    } = reClammConfigForm.watch()
+
+    const minPrice = areTokensInOrder ? initialMinPrice : invertNumber(initialMaxPrice)
+    const maxPrice = areTokensInOrder ? initialMaxPrice : invertNumber(initialMinPrice)
+    const targetPrice = areTokensInOrder ? initialTargetPrice : invertNumber(initialTargetPrice)
```

Comment:
then if tokens are out of order in LS state we simply invert the params behind the scenes in the create pool input logic